datafusion/src/catalog/schema.rs

```diff
@@ -57,6 +57,14 @@ pub trait SchemaProvider: Sync + Send {
             "schema provider does not support deregistering tables".to_owned(),
         ))
     }
+
+    /// If supported by the implementation, checks the table exist in the schema provider or not.
+    /// If no matched table in the schema provider, return false.
+    /// Otherwise, return true.
+    #[allow(unused_variables)]
+    fn table_exist(&self, name: &str) -> bool {
```

Comment:
this default implementation doesn't seem very useful, if we require users to implement `fn table_names`, then i think it's reasonable to require them to implement `table_exists` as well.

Reply:
agree with your suggestion.